src/components/IstioWizards/IstioWizard.tsx

```diff
@@ -99,15 +85,44 @@ class IstioWizard extends React.Component<Props, State> {
     return true;
   };
 
-  componentDidMount() {
-    this.resetState();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (prevProps.show !== this.props.show || !this.compareWorkloads(prevProps.workloads, this.props.workloads)) {
-      this.resetState();
+  buildHTTPMatchRequest = (matches: string[]): HTTPMatchRequest[] => {
+    const matchRequests: HTTPMatchRequest[] = [];
+    const matchHeaders: HTTPMatchRequest = { headers: {} };
+    // Headers are grouped
+    matches
+      .filter(match => match.startsWith('headers'))
+      .forEach(match => {
+        // match follows format:  headers [<header-name>] <op> <value>
+        const i0 = match.indexOf('[');
+        const j0 = match.indexOf(']');
+        const headerName = match.substring(i0 + 1, j0);
+        const i1 = match.indexOf(' ', j0 + 1);
+        const j1 = match.indexOf(' ', i1 + 1);
+        const op = match.substring(i1 + 1, j1);
+        const value = match.substring(j1 + 1);
+        matchHeaders.headers![headerName] = { [op]: value };
```

Comment:
My proposal here is get values with regexlike something like this. What do you think?
```ts
const reg = /headers \[([a-z0-9\-]+)\] ([a-z0-9\-]+) ([a-z0-9\-]+)/;
const result =match.match(namespaceRegex) || [];
const headerName = result[1]
const op = result[2]
const value = result[3]
```

Reply:
Regex are interesting, but main blocker/pain I see is that value in third param can be also a regex, so, having a regex as parameter could lead in a more complex value.
As empty spaces are fixed and internal to that variable I found more easy to "parse" for this particular use case.

Reply:
ok !!! I understand